tailwind-plugins/form-control.js

```diff
@@ -0,0 +1,181 @@
+      border: `1px solid ${DANGER}`,
+    },
+    ".form-control-feedback": {
+      color: DANGER,
+    },
+    ".dark &": {
+      ".form-control": {
+        borderColor: DARK_DANGER,
+        borderWidth: "2px",
+      },
+      ".form-control-feedback": {
+        color: DARK_DANGER,
+      },
+    },
+  };
+}
+
+/**
+ * Stylings for select.form-control (converted from Bootstrap 4.6 with our custom SCSS)
+ * @param {*} theme Tailwind theme config
+ * @returns {object} Stylings
+ */
+function selectFormControl(theme) {
+  return {
+    appearance: "none",
+    backgroundRepeat: "no-repeat",
+    backgroundPosition: `right ${SELECT.paddingX} top 50%`,
+    backgroundImage: 'url("../_images/glyphs/down-chevron.svg")',
+    backgroundSize: `${SELECT.backgroundSize} ${SELECT.backgroundSize}`,
+    border: `1px solid ${theme("colors.gray.20")}`,
+    borderRadius: "0",
+    color: theme("colors.black"),
+    padding: `5px ${SELECT.paddingRight} 5px ${SELECT.paddingX}`,
+    ".dark &": {
+      backgroundImage: "url(../_images/glyphs/down-chevron-dark-theme.svg)",
+      backgroundColor: "transparent",
+      border: `1px solid ${theme("colors.white")}`,
+      color: theme("colors.white"),
+      option: {
+        color: theme("colors.black"),
+      },
+      "&:focus": {
+        color: theme("colors.white"),
+        backgroundColor: "transparent",
+        borderColor: theme("colors.white"),
+      },
+    },
+    "&:focus::-ms-value": {
+      // Suppress the nested default white text on blue background highlight given to
+      // the selected option text when the (still closed) <select> receives focus
+      // in IE and (under certain conditions) Edge, as it looks bad and cannot be made to
+      // match the appearance of the native widget.
+      // See https://github.com/twbs/bootstrap/issues/19398.
+      color: "#495057",
+      backgroundColor: theme("colors.white"),
+    },
+  };
+}
+
+/**
+ * Stylings for select.form-control-lg (converted from Bootstrap 4.6 with our custom SCSS)
+ * @param {*} theme Tailwind theme config
+ * @returns {object} Stylings
+ */
+function selectFormLgControl(theme) {
+  return {
+    ...selectFormControl(theme),
+    height: "auto",
+    padding: `${SELECT.paddingX} ${SELECT.paddingRight} ${SELECT.paddingX} ${SELECT.paddingX}`,
+    fontSize: "1.25rem",
+    lineHeight: "1.25",
+  };
+}
+
+module.exports = [
+  plugin(function ({ addComponents, theme }) {
+    addComponents([
+      {
+        ".form-control": {
+          ...formControl(theme),
+        },
+        "select.form-control": {
+          ...selectFormControl(theme),
+        },
+        "select.form-control-lg": {
+          ...selectFormLgControl(theme),
+        },
+        "input[type='date'].form-control, input[type='time'].form-control, input[type='datetime-local'].form-control, input[type='month'].form-control":
+          {
+            appearance: "none",
+          },
+        ".has-danger": {
+          ...hasDanger(theme),
```

Comment:
You can address the `.has-danger` class in two different ways. You can make a danger variant and address every HTML instance of `form-control` and utility classes. Or you can address has-danger the same way we kinda address adding dark class for form-control in L117. I do not think the has-danger function work in this case.

Reply:
Thanks for the suggestion. Can you elaborate on `I do not think the has-danger function work in this case.`? I checked again and verified the `.tw-has-danger .form-control` and `.tw-has-danger .form-control-feedback` DOM elements are having the stylings I defined here.

Reply:
![image](https://user-images.githubusercontent.com/2374073/223272985-58367ccf-c532-4f67-ab16-79dbe18ad159.png)
When you hover over `.tw-form-control` it reveals zero information about the has-danger classes. 

Hovering over `tw-has-danger` does nothing as well and should not be a component.

Reply:
@fessehaye I think the above problem is actually from Tailwind CSS IntelliSense. I tried to comment out `...formControlPlugins,` in `tailwind.config.js` and uncomment it again, everything then got generated as expected.

<img width="419" alt="image" src="https://user-images.githubusercontent.com/2896608/223648110-1b7cb695-4245-485f-b85c-27c19c23249f.png">

Reply:
interesting, I guess since we are moving away from this implementation of has-danger, it doesn't matter atm